web/modules/custom/joinup_user/src/Form/UnsubscribeFromAllCollectionsForm.php

```diff
@@ -0,0 +1,281 @@
+   */
+  protected $membershipManager;
+
+  /**
+   * The renderer service.
+   *
+   * @var \Drupal\Core\Render\Renderer
+   */
+  protected $renderer;
+
+  /**
+   * The user entity from the route.
+   *
+   * @var \Drupal\Core\Session\AccountInterface
+   */
+  protected $user;
+
+  /**
+   * Constructs a UnsubscribeFromAllCollectionsForm.
+   *
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager service.
+   * @param \Drupal\og\MembershipManagerInterface $membership_manager
+   *   The membership manager service.
+   * @param \Drupal\Core\Render\Renderer $renderer
+   *   The renderer service.
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The current route match.
+   */
+  public function __construct(EntityTypeManagerInterface $entity_type_manager, MembershipManagerInterface $membership_manager, Renderer $renderer, RouteMatchInterface $route_match) {
+    $this->entityTypeManager = $entity_type_manager;
+    $this->membershipManager = $membership_manager;
+    $this->renderer = $renderer;
+    $this->user = $route_match->getParameter('user');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('entity_type.manager'),
+      $container->get('og.membership_manager'),
+      $container->get('renderer'),
+      $container->get('current_route_match')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'unsubscribe_from_all_collections_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getQuestion() {
+    return $this->t('Unsubscribe from all collections');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDescription() {
+    return $this->t('Are you sure you want to unsubscribe from all collections?<br />You will stop receiving news and updates from all collections, including those you are a facilitator in.');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getCancelUrl() {
+    return Url::fromRoute('joinup_subscription.subscriptions_page');
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    if ($memberships = $this->getUserMembershipIds()) {
+      $memberships = $this->entityTypeManager->getStorage('og_membership')->loadMultiple($memberships);
+      $labels = array_map(function (OgMembershipInterface $membership) {
+        return $membership->getGroup()->label();
+      }, $memberships);
+      asort($labels);
+      $form = parent::buildForm($form, $form_state);
+      $form['information'] = [
+        '#type' => 'item',
+        '#markup' => t('You are currently registered to be notified for the following collections:'),
+        '#tree' => TRUE,
+        'items' => [
+          '#theme' => 'item_list',
+          '#items' => $labels,
+        ],
+      ];
+    }
+    else {
```

Comment:
If we go with the UX change, this should throw an exception because such a case should not exist and reveals a development problem.

Reply:
From your suggestions above, I will add the access check in the form itself as well. That means that even by entering the url, there will be no access there. Which means that this is not needed anymore.